### DISCUSSION TITLE
Is there any related detailed design for arctic use iceberg V2  to rewrite

### DISCUSSION THREAD

**Discussion by huyuanfeng2018** [apache/amoro#1484] [Q&A]
![image](https://github.com/NetEase/arctic/assets/40817998/aabf3872-351c-482b-bdd6-2a97942e6ddd)
![image](https://github.com/NetEase/arctic/assets/40817998/6e3b37db-6b92-4282-9a15-84c9c8f4632e)

I Want to know how to delete the delete file

**Comment by zhoujinsong** [comment]
Hi, @huyuanfeng2018 
After equality delete file is rewritten as positional delete file, it can be safely removed from the table. The deletion operation can be completed through the RewriteFiles interface provided by Iceberg. However, during the deletion process, it is possible to encounter a situation where the delete file to be deleted is no longer in the table, in which case the exception can be ignored.

You can find the relevant code here：https://github.com/NetEase/arctic/blob/refactor-ams/ams/server/src/main/java/com/netease/arctic/server/optimizing/UnKeyedTableCommit.java#L213

**Reply by huyuanfeng2018** [reply]
thx😁
